### DISCUSSION TITLE
[refinementList] New option to keep current search after click

### DISCUSSION THREAD

**Discussion by cmalard** [algolia/instantsearch#5206] [Ideas]
**Do you want to request a *feature* or report a *bug*?**
feature

**Feature: What is your use case for such a feature?**
- I want to select many results that start the same
- I have to write them each time because the search is cleared
![screen](https://user-images.githubusercontent.com/1692136/32560515-1ff8a1fc-c4aa-11e7-9d4a-70c921b0922e.gif)

**Feature: What is your proposed API entry? The new option to add? What is the behavior?**
Add an option to keep the search after a click

**What is the version you are using? Always use the latest one before opening a bug issue.**
2.2

**Comment by bobylito** [comment]
Thanks @cmalard for this feature proposal. From what I understand, being able to select multiple using a single search of facet values would make sense. Do you have time to dig in the code and contribute a PR for this new feature?

**Comment by cmalard** [comment]
Not for now, but as soon as it is available I will use it ! :-)

**Comment by bobylito** [comment]
Ok good to know. Thanks :)

**Comment by bobylito** [comment]
Btw @mthuret you might be interested in that :)

**Comment by mthuret** [comment]
@bobylito thanks for notifying me! Indeed I think that would be a great addition to our widgets having search for facets.

**Comment by u3804** [comment]
Hi All!  In addition to seconding the request of @cmalard, here's a related use case.  Feel free to disregard this if you don't think others would benefit.

1. **Example**: search "insignia" in the facet search box at [https://codesandbox.io/s/5yr601pk2p](url)
![image](https://user-images.githubusercontent.com/5862715/42134842-7b3f4ad8-7d60-11e8-9e73-3ce901da4613.png)

_Per the feature request above, we'd be able to click on each of the three facets without having to search again._

2. **Add-on feature requests**:

- would adding a mechanism to select all of the facet search results be possible?  In the example above, that would mean selecting all three facets matching "insignia" without having to click three times.  **_I ask because my facet data has a very long tail of data that has not been standardized (i.e. many relevant results per likely facet search term)._** 

- would providing an option to revise the behavior after a user hits `enter` be possible?  right now, hitting `enter` retrieves the top facet result.  Maybe in some circumstances retrieving all matching results might be a better fit?

**Comment by bobylito** [comment]
Thanks for your input, we're not going to work on that right away but I've notified the product team :)

**Comment by ghost** [comment]
Any progress on this?

**Reply by Haroenv** [reply]
No, there hasn't been any further demand for this.
